### PR TITLE
Loader tool: Refactor loader tool (for AYON)

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Changelog Description
Refactored loader tool to new tool. Separated backend and frontend logic. Refactored logic is AYON-centric and is used only in AYON mode, so it does not affect OpenPype. The tool is also replacing library loader.

## Additional info
There were made smaller changes in the tool UI. Versions comboboxes are visible all the time, the tool can be used as both loader and library loader tools. The tool does not contain logic related to site sync - that will be added in future PRs.

### Screenshot
image

## Known issues/TODOs
- [ ] Missing docstring (everywhere).
- [x] Loader plugins do not recognize if are shown in "library loader" mode or "loader" mode.
- [x] Missing option to change product group (Ctrl + G shortcut).
- [x] Not used in tray action yet.
- [x] Find more suitable size of the window.
- ... report anything you'll hit

### Basic tests
1. Create openpype package
2. Update openpype package to server
3. Set the package to production bundle
4. Run AYON-launcher
5. Run a host of your choise (should have loader tool)
6. Try to use the loader tool
7. It should work as expected
Possible usecases:
- load should work
- crashed load should show message box
- load options should show UI which is propagating the values to load
- refresh should refresh
- grouping should show groups
- thumbnails should be visible
- all kind of filtering should filter

### CODE REVIEW

**Comment by BigRoy** [comment]
Could you add a screenshot of the current state of the 'redesign'? (Just so that, without testing, one could also see the always available comboboxes for example)

**Comment by LiborBatek** [comment]
When triggering ```Reference Abc``` or ```Reference Ma``` in both cases the ```Loader``` will load both repre...no matter which one was selected...should be just the one selected.
image

**Comment by LiborBatek** [comment]
Seems like its trying to load all repre available on asset even when selecting just particular type ...in this case ```abc``` ```fbx``` ```max``` etc...
image

**Comment by BigRoy** [comment]
By the way, is this lacking the **Dependency** tab?

**Comment by iLLiCiTiT** [comment]
> By the way, is this lacking the Dependency tab?

Oh yes, forgot about dependency tab. Yes, it's not there.

**Comment by BigRoy** [comment]
And I know it's not up to this PR to add new features - but I would 100% love there to be a Versions tabs that is basically a list widget showing the publish dates, status, author and comments of all versions for the selected subset.

Currently there's just no UI that can show you the publish dates of the last few versions and their respective comments.

Here's a Prism equivalent on the right hand side:

_Not super pretty but it shows the data I'm referring to._

I thought we had an existing issue to track that as a feature request - but I can't find it.

**Comment by BigRoy** [comment]
> When thumbnail having transparency it should show solid bg color and probably not checkered bg
> 
> image

Or at the very least have a less harsh checkboard pattern, lighter gray and white can help:



How come the checkerboard appears to be 'red tinted' - do you have some screen tools to dim your screens (like "night light" or alike?)

**Comment by LiborBatek** [comment]
> How come the checkerboard in your screenshot appears to be 'red tinted' - do you have some screen tools to dim your screens (like "night light" or alike?)

Nope. Strange...

I think it already been handled (transparency with thumbnails) in Legacy OP as seen here...I really like it comped onto bg color btw, looks neat!


https://github.com/ynput/OpenPype/assets/112623825/6a4f9da5-8614-4cb6-8e53-0d95fdd4d032

File: openpype/tools/ayon_loader/models/actions.py [hunk 1]
```diff
@@ -0,0 +1,870 @@
...
+        for loader in repre_loaders:
+            # # do not allow download whole repre, select specific repre
+            # if tools_lib.is_sync_loader(loader):
+            #     continue
+
+            for repre_name, repre_contexts in repre_contexts_by_name.items():
+                filtered_repre_contexts = filter_repre_contexts_by_loader(
+                    repre_contexts, loader)
+                if len(filtered_repre_contexts) != len(repre_contexts):
+                    continue
```
Comment (BigRoy) [inline openpype/tools/ayon_loader/models/actions.py:593]: Are we sure this is correct - technically a representation that matches _only by name_ might have on representation be valid for a loader and another won't.

There could be two representations named `ma` but the loader could be filtered out by `family`
There could be two representations named `render` but the loader could be filtered out by `extensions`

I would expect the continue statement to be:

```suggestion
                if not filtered_repre_contexts:
                    continue
```

Or am I misreading this code?

Comment (iLLiCiTiT) [reply openpype/tools/ayon_loader/models/actions.py:593]: You're right, I don't know why it was there TBH.

**Review by LiborBatek — APPROVED** [review]
Did multiple tests in different scenarios and all seem working normally aka well.

**Pull request merged**